packages/react/src/wishlists/hooks/useWishlist.ts

```diff
@@ -3,105 +3,78 @@
  * wishlists.
  */
 import {
-  addWishlistItem as addWishlistItemAction,
-  fetchWishlist as fetchWishlistAction,
+  addWishlistItem,
+  fetchWishlist,
+  getUser,
   getWishlist,
   getWishlistError,
-  getWishlistId,
   getWishlistItems,
-  getWishlistItemsCounter,
-  getWishlistTotalQuantity,
-  isWishlistLoading as isWishlistLoadingSelector,
-  resetWishlist as resetWishlistAction,
-  resetWishlistState as resetWishlistStateAction,
+  isWishlistLoading,
+  removeWishlistItem,
+  resetWishlist,
+  updateWishlistItem,
 } from '@farfetch/blackout-redux';
 import { useAction } from '../../helpers';
+import { useEffect } from 'react';
 import { useSelector } from 'react-redux';
-import type { UseWishlist } from './types';
+// Types
+import type {
+  AddItem,
+  FetchWishlist,
+  Options,
+  RemoveItem,
+  ResetWishlist,
+  UpdateItem,
+} from './types';
 
 /**
  * Provides Redux actions and state access, as well as handlers for dealing with
  * wishlist business logic.
  *
  * @returns All the handlers, state, actions and relevant data needed to manage any wishlist operation.
  */
-const useWishlist: UseWishlist = () => {
+const useWishlist = (options: Options = {}) => {
+  const { enableAutoFetch = false } = options;
   // Selectors
   const error = useSelector(getWishlistError);
-  const id = useSelector(getWishlistId);
-  const isWishlistLoading = useSelector(isWishlistLoadingSelector);
-  const itemCount = useSelector(getWishlistItemsCounter);
+  const isLoading = useSelector(isWishlistLoading);
   const items = useSelector(getWishlistItems);
-  const itemsWithSetsInfo = useSelector(state => getWishlistItems(state, true));
-  const totalQuantity = useSelector(getWishlistTotalQuantity);
   const wishlist = useSelector(getWishlist);
+  const userWishlistId = useSelector(getUser)?.wishlistId;
   // Actions
-  const addWishlistItem = useAction(addWishlistItemAction);
-  const fetchWishlist = useAction(fetchWishlistAction);
-  const resetWishlist = useAction(resetWishlistAction);
-  const resetWishlistState = useAction(resetWishlistStateAction);
+  const addItem: AddItem = useAction(addWishlistItem);
+  const updateItem: UpdateItem = useAction(updateWishlistItem);
+  const removeItem: RemoveItem = useAction(removeWishlistItem);
+  const fetch: FetchWishlist = useAction(fetchWishlist);
+  const reset: ResetWishlist = useAction(resetWishlist);
   // Data with some logic
-  const isEmpty = totalQuantity === 0;
-  // If there is no wishlist and there is no error (with `isWishlistLoading === false`),
-  // it means the wishlist hasn't started fetching, so it's considered loading
-  const isLoading = (!wishlist && !error) || isWishlistLoading;
+  const count = wishlist.count;
+  const isEmpty = count === 0;
+  const isFetched = !!wishlist.id && typeof count !== 'undefined' && count >= 0;
```

Comment:
This validation is necessary since TS throws an error identifying `count` as possibly `undefined`. If you prefere I can change it to `const isFetched = !!wishlist.id && !!count && count >= 0;`

Reply:
Actually, since after fetch we have both wishlist id and count, we don't need to validate the count value. 
Updated

Reply:
Nice! Thanks!